src/cli.rs

```diff
@@ -10,25 +10,23 @@ use crate::tap;
 use crate::top;
 use crate::{config, generate, get_version, graph, list, unit_test, validate};
 
-#[derive(StructOpt, Debug)]
-#[structopt(rename_all = "kebab-case")]
+#[derive(Parser, Debug)]
+#[clap(rename_all = "kebab-case")]
 pub struct Opts {
-    #[structopt(flatten)]
+    #[clap(flatten)]
     pub root: RootOpts,
 
-    #[structopt(subcommand)]
+    #[clap(subcommand)]
     pub sub_command: Option<SubCommand>,
 }
 
 impl Opts {
     pub fn get_matches() -> Self {
         let version = get_version();
-        let app = Opts::clap().version(version.as_str()).global_settings(&[
-            AppSettings::ColoredHelp,
-            AppSettings::InferSubcommands,
-            AppSettings::DeriveDisplayOrder,
-        ]);
-        Opts::from_clap(&app.get_matches())
+        let app = Opts::command()
+            .version(version.as_str())
+            .global_setting(AppSettings::DeriveDisplayOrder);
+        Opts::from_arg_matches(&app.get_matches()).expect("couldn't initialize CLI. Please report.")
```

Comment:
`Opts::from_clap` is deprecated in favour of `Opts::from_arg_matches`, which now returns `Result<Self, Error>`.

I'm not 100% on which conditions would cause this to panic, other than misconfigured field annotations, so I've deemed this an 'us' problem and elected to panic with a message to report.

Reply:
Yeah, that method doesn't document itself very well. Looking at `clap::Error` the kinds are these:

https://docs.rs/clap/latest/clap/error/enum.ErrorKind.html

Which makes me think that maybe this would return an error simply if the user passed, for example, an invalid subcommand? In which case, we should propagate it up?

If we do choose to panic here, it'd be nice to include the `Err` value itself to have more of a clue of what failed.

Reply:
Ah, thanks for looking -- yeah, propagating makes sense.

Adjusted in https://github.com/vectordotdev/vector/pull/11470/commits/c6a1da9e9a20c93db3de176c1bd31f63aa2454b4 to print a [friendlier error](https://docs.rs/clap/latest/src/clap/error/kind.rs.html#401-437) and return `exitcode::USAGE`